ppl-spark-integration/src/main/java/org/opensearch/sql/ast/tree/AppendCol.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.sql.ast.tree;
+
+import com.google.common.collect.ImmutableList;
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+import org.opensearch.sql.ast.AbstractNodeVisitor;
+import org.opensearch.sql.ast.Node;
+
+import java.util.List;
+
+/**
+ * A composite object which store the subQuery along with some more ad-hoc option like override
+ */
+@Getter
+@Setter
+@ToString
+@EqualsAndHashCode(callSuper = false)
+@AllArgsConstructor
+public class AppendCol extends UnresolvedPlan {
+
+    public boolean override;
+
+    private UnresolvedPlan subSearch;
+
+    private UnresolvedPlan child;
```

Comment:
Can these two attributes be `final`?

```suggestion
    public final boolean override;

    private final UnresolvedPlan subSearch;

    private UnresolvedPlan child;
```

Reply:
Agree, as both variables should be effective final, I have updated the visiter class to use setter, instead of direct access.